x-pack/plugins/ml/public/application/components/controls/select_interval/select_interval.test.tsx

```diff
@@ -5,58 +5,39 @@
  * 2.0.
  */
 
-import React from 'react';
-import { act } from 'react-dom/test-utils';
-import { MemoryRouter } from 'react-router-dom';
-import { mount } from 'enzyme';
-
-import { EuiSelect } from '@elastic/eui';
-
-import { UrlStateProvider } from '@kbn/ml-url-state';
+import React, { useState } from 'react';
+import { render, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 import { SelectInterval } from './select_interval';
 
-describe('SelectInterval', () => {
-  test('creates correct initial selected value', () => {
-    const wrapper = mount(
-      <MemoryRouter>
-        <UrlStateProvider>
-          <SelectInterval />
-        </UrlStateProvider>
-      </MemoryRouter>
-    );
-    const select = wrapper.find(EuiSelect);
+// The following mock setup is necessary so that we can simulate
+// both triggering the update callback and the internal state update
+// to update the dropdown to the new state.
+const mockUpdateCallback = jest.fn();
+const mockUseState = jest.fn().mockImplementation(useState);
+jest.mock('@kbn/ml-url-state', () => ({
+  usePageUrlState: () => {
+    const [interval, setInterval] = mockUseState({ display: 'Auto', val: 'auto' });
+    return [interval, mockUpdateCallback.mockImplementation((d) => setInterval(d))];
+  },
+}));
 
-    const defaultSelectedValue = select.props().value;
-    expect(defaultSelectedValue).toBe('auto');
-  });
-
-  test('currently selected value is updated correctly on click', (done) => {
-    const wrapper = mount(
-      <MemoryRouter>
-        <UrlStateProvider>
-          <SelectInterval />
-        </UrlStateProvider>
-      </MemoryRouter>
-    );
-    const select = wrapper.find(EuiSelect).first();
-    const defaultSelectedValue = select.props().value;
-    expect(defaultSelectedValue).toBe('auto');
+describe('SelectInterval', () => {
+  it('updates the selected value correctly on click', () => {
+    // prepare
```

Comment:
nit: it's good to have such comments! but usually, it's referred to as Arrange, Act and Assert (AAA) Pattern.